Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h

```diff
@@ -183,6 +185,8 @@ class XMLHttpRequest final : public XMLHttpRequestEventTarget {
     //     NOTE: This needs to be a JS::Value as the JSON response might not actually be an object.
     Variant<JS::Value, Failure, Empty> m_response_object;
 
+    JS::GCPtr<DOM::Document> m_response_document; // FIXME: remove this and use m_response_object
```

Comment:
if `XMLHttpRequest::set_document_response()` puts document into `m_response_object` it should be possible to extract document from there once it is needed instead of introducing `m_response_document`.

Reply:
Yes sure but I couldn't get the to work, if you have a code snippet of the cast I will update the commit

Reply:
actually it seems you don't ever need to extract document so it should be ok to just delete it?

Reply:
oh, sorry, I see `response_xml()`

Reply:
looks correct?
```diff
diff --git a/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp b/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp
index 2d9be46b5e..257fa19900 100644
--- a/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp
+++ b/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp
@@ -138,13 +138,13 @@ WebIDL::ExceptionOr<JS::GCPtr<DOM::Document>> XMLHttpRequest::response_xml()

     // 4. If this’s response object is non-null, then return it.
     if (!m_response_object.has<Empty>())
-        return m_response_document;
+        return &verify_cast<DOM::Document>(m_response_object.get<JS::Value>().as_object());

     // 5. Set a document response for this.
     set_document_response();

     // 6. Return this’s response object.
-    return m_response_document;
+    return &verify_cast<DOM::Document>(m_response_object.get<JS::Value>().as_object());
 }

 // https://xhr.spec.whatwg.org/#dom-xmlhttprequest-responsetype
@@ -337,7 +337,6 @@ void XMLHttpRequest::set_document_response()

     // 12. Set xhr’s response object to document.
     m_response_object = JS::Value(document);
-    m_response_document = document;
 }

 // https://xhr.spec.whatwg.org/#final-mime-type
diff --git a/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h b/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h
index ea89f2ea44..3f7a481752 100644
--- a/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h
+++ b/Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h
@@ -185,8 +185,6 @@ private:
     //     NOTE: This needs to be a JS::Value as the JSON response might not actually be an object.
     Variant<JS::Value, Failure, Empty> m_response_object;

-    JS::GCPtr<DOM::Document> m_response_document; // FIXME: remove this and use m_response_object
-
     // https://xhr.spec.whatwg.org/#xmlhttprequest-fetch-controller
     // fetch controller
     //     A fetch controller, initially a new fetch controller.
```

Reply:
Yes thank you 😃